common/app/services/NewsletterService.scala

```diff
@@ -49,6 +51,17 @@ class NewsletterService(newsletterSignupAgent: NewsletterSignupAgent) {
     }
   }
 
+  private def getNewsletterResponseFromSignUpPage(articleId:String): Option[NewsletterResponse] = {
+    newsletterSignupAgent.getNewsletters() match {
+      case Left(_) => None
+      case Right(list) => list.find(response => response.signupPage.nonEmpty && response.signupPage.get == "/"+articleId)
+    }
+  }
+
+  private def isSignUpPage(articlePage: ArticlePage): Boolean = {
+    articlePage.article.content.metadata.format.nonEmpty && articlePage.article.content.metadata.format.get.design == NewsletterSignupDesign
```

Comment:
I think you can do this instead on an Option 

```suggestion
    articlePage.article.content.metadata.format.design.contains(NewsletterSignupDesign)
```

Reply:
It doesn't like that - maybe because its the "format" that is an `Option`, not "design"? 
I tried a few changes and this seems to fit with the style better than what I had before:

```scala
  private def isSignUpPage(articlePage: ArticlePage): Boolean = {
    articlePage.article.content.metadata.format match {
      case None         => false
      case Some(format) => format.design == NewsletterSignupDesign
    }
  }
```